connection.go

```diff
@@ -144,6 +144,10 @@ func (mc *mysqlConn) interpolateParams(query string, args []driver.Value) (strin
 	buf = buf[:0]
 	argPos := 0
 
+	if len(args) != strings.Count(query, "?") {
+		return "", ErrArgsInvalid
```

Comment:
I checked MySQL reference.  ? can't appear in quotes.
So checking here is OK, while I think checking just before `arg := args[argPos]` is safer.



> And i don't understand why drvier.ErrSkip safer

`database/sql` fallbacks to real prepared statement when driver.ErrSkip is returned.